cypress/integration/unit_tests/color_function_util_test.js

```diff
@@ -79,23 +182,83 @@ describe('Unit tests for color function utility', () => {
+    setDisaster('2017-harvey');
+    readDisasterDocument().then((doc) => {
+      const {layerArray} = doc.data();
+      let featureCollectionLayer;
+      for (const layer of layerArray) {
+        if (layer.assetType === LayerType.FEATURE_COLLECTION) {
+          featureCollectionLayer = layer;
+          break;
+        }
+      }
+      if (!featureCollectionLayer) {
+        // If harvey ends up with no feature collection-typed layers we
+        // return early. Don't expect this to happen but.
+        return;
+      }
+
+      const {colorFunction} = featureCollectionLayer;
+      const {color, property, colors} = colorFunction;
+
+      const td = setUpWithLayer(featureCollectionLayer);
+      td.trigger('click');
+
+      switch (colorFunction.currentStyle) {
+        case ColorStyle.CONTINUOUS:
+          expect($('#property-picker').val()).to.equal(property);
+          expect($('#continuous-min').val())
+              .to.equal(colorFunction.columns[property].min);
+          expect($('#continuous-max').val())
+              .to.equal(colorFunction.columns[property].max);
+          if (color) {
+            expect($('#continuous-color-picker').val()).to.equal(color);
+          } else {
+            expect($('#continuous-color-picker').val()).to.be.null;
+          }
+          break;
+        case ColorStyle.DISCRETE:
+          expect($('#property-picker').val()).to.equal(property);
+          $('#discrete-color-pickers')
+              .find('li')
+              .each(/* @this HTMLElement */ function() {
+                // slicing off the ': ' at the end of the label to just get the
+                // property value.
+                const value = $(this).children('label').text().slice(0, -2);
+                const selectedColor = $(this).children('select').val();
+                expect(colors[value]).to.equal(selectedColor);
+              });
+          break;
+        case ColorStyle.SINGLE:
+          if (color) {
+            expect($('#single-color-picker').val()).to.equal(color);
+          } else {
+            expect($('#single-color-picker').val()).to.be.null;
+          }
+          break;
+      }
+    });
+  });
+
   it('switches schemas and writes data', () => {
     const layer = {
-      'colorFunction': {
-        'currentStyle': 2,
-        'columns': {
-          'wings': {'min': 0, 'max': 2, 'values': [0, 1, 2]},
-          'legs': {'min': 0, 'max': 100, 'values': [0, 2, 4, 8, 100]},
+      colorFunction: {
+        currentStyle: 2,
+        lastByPropertyStyle: 0,
+        columns: {
+          wings: {min: 0, max: 2, values: [0, 1, 2]},
+          legs: {min: 0, max: 100, values: [0, 2, 4, 8, 100]},
         },
-        'colors': {},
-        'color': 'yellow',
+        colors: {},
+        color: 'yellow',
       },
     };
     const td = setUpWithLayer(layer);
-    expect(colorFunctionEditor.is(':visible')).to.be.false;
-
+    expect($(colorFunctionEditor).is(':visible')).to.be.false;
```

Comment:
I think Cypress chai has jQuery assertions, so even if you're not using Cypress selectors, you can get more meaningful assertions: https://docs.cypress.io/guides/references/assertions.html#Chai-jQuery